### ISSUE TITLE
flutter upgrade issue

### ISSUE THREAD

**Issue by MontrealMike** [flutter/flutter#114840]
ubuntu 22.10

running the command flutter upgrade yields the following output:
```
michael@meerkat-2$ flutter upgrade
clang++: symbol lookup error: clang++: undefined symbol: , version LLVM_10
Downloading Linux x64 Dart SDK from Flutter engine 857bd6b74c5eb56151bfafe91e7fa6a82b6fee25...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  204M  100  204M    0     0  9856k      0  0:00:21  0:00:21 --:--:-- 10.1M
unzip: error while loading shared libraries: /snap/core20/current/lib/x86_64-linux-gnu/libbz2.so.1.0: unsupported version 0 of Verneed record

It appears that the downloaded file is corrupt; please try again.
If this problem persists, please report the problem at:
  https://github.com/flutter/flutter/issues/new?template=1_activation.md
```

I have tried a few times with the same result.

flutter doctor yields:
```
michael@meerkat-2$ flutter doctor -v
clang++: symbol lookup error: clang++: undefined symbol: , version LLVM_10
clang++: symbol lookup error: clang++: undefined symbol: , version LLVM_10
Downloading Linux x64 Dart SDK from Flutter engine 857bd6b74c5eb56151bfafe91e7fa6a82b6fee25...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  204M  100  204M    0     0  10.3M      0  0:00:19  0:00:19 --:--:-- 10.3M
unzip: error while loading shared libraries: /snap/core20/current/lib/x86_64-linux-gnu/libbz2.so.1.0: unsupported version 0 of Verneed record

It appears that the downloaded file is corrupt; please try again.
If this problem persists, please report the problem at:
  https://github.com/flutter/flutter/issues/new?template=1_activation.md
```

Note the clang++ message.  I am not sure whether this is connected to the upgrade problem.

**Comment by huycozy** [comment]
Hi @MontrealMike 
What is your previous Flutter version? Can you share the output of `uname -r`?
Also, it seems you have installed Flutter with `snapd`, right? If so, please check this issue on snap forum: https://forum.snapcraft.io/t/unsupported-version-0-of-verneed-record-linux-6-0/32160 (Should be fixed with linux 6.0.7). You can also check a workaround at [here](https://forum.snapcraft.io/t/unsupported-version-0-of-verneed-record-linux-6-0/32160/16).

**Comment by MontrealMike** [comment]
Thank you.

Here is my uname -r:
***@***.***$ uname -r
6.0.2-76060002-generic

I will look into the workaround you referenced.
Regards,
Mike

------------------------------

Michael Bleau | ***@***.*** | 514 247-2944
